dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckDataRelationsValidationHook.java

```diff
@@ -116,20 +117,39 @@ public void validateEnrollment( ValidationErrorReporter reporter, Enrollment enr
         }
     }
 
+    private boolean programDoesNotHaveOrgUnit( Program program, OrganisationUnit orgUnit,
+        Map<String, List<String>> programAndOrgUnitsMap )
+    {
+        return !programAndOrgUnitsMap.containsKey( program.getUid() )
+            || !programAndOrgUnitsMap.get( program.getUid() ).contains( orgUnit.getUid() );
+    }
+
     @Override
     public void validateEvent( ValidationErrorReporter reporter, Event event )
     {
         TrackerImportValidationContext context = reporter.getValidationContext();
-
         ProgramStage programStage = context.getProgramStage( event.getProgramStage() );
         OrganisationUnit organisationUnit = context.getOrganisationUnit( event.getOrgUnit() );
         Program program = context.getProgram( event.getProgram() );
 
+        validateProgramStageInProgram( reporter, event, programStage, program );
+        validateRegistrationProgram( reporter, event, context, program );
+        validateProgramHasOrgUnit( reporter, event, context, organisationUnit, program );
+        validateEventCategoryCombo( reporter, event, program );
```

Comment:
Would make sense to have the validation of category combo in its own hook?

Reply:
Maybe 🤷🏻 There is also EventCategoryOptValidationHook which already interacts with PreCheckDataRelationsValidationHook. PreCheckDataRelationsValidationHook has to come before EventCategoryOptValidationHook. PreCheckDataRelationsValidationHook has removeOnError `true` while `EventCategoryOptValidationHook` does not.

I could create a EventCategoryComboValidationHook with removeOnError `true` and make sure it runs before EventCategoryOptValidationHook. I just don't know if that has any effect on other validation hooks 🤷🏻 
If someone feels strongly about it I can investigate this in this issue. Otherwise I would postpone it, since this is already quite big of a change.

Reply:
Let's postpone it, absolutely.
It was just a consideration